Framework/Source/Utils/Gui.cpp

```diff
@@ -279,15 +280,20 @@ namespace Falcor
     bool Gui::addCheckboxes(const char label[], bool* pData, uint32_t numCheckboxes, bool sameLine)
     {
         bool modified = false;
-        std::string labelString(label);
-        labelString.append("[0]");
+        std::string labelString(std::string("##") + label);
+        labelString.push_back('0');
 
-        for (uint32_t i = 0; i < numCheckboxes; ++i)
+        for (uint32_t i = 0; i < numCheckboxes - 1; ++i)
         {
-            labelString[labelString.size() - 2] = '0' + static_cast<int32_t>(i);
-            modified |= addCheckBox(labelString.c_str(), pData[i], sameLine);
+            labelString[labelString.size() - 1] = '0' + static_cast<int32_t>(i);
+            modified |= addCheckBox(labelString.c_str(), pData[i], (!i) ? sameLine : true);
         }
 
+        ImVec2 newCursorPosition = ImGui::GetCursorScreenPos();
+        newCursorPosition.x += ImGui::GetContentRegionAvail().x;
+        ImGui::SetCursorScreenPos(newCursorPosition);
+        addCheckBox(label, pData[numCheckboxes - 1], (numCheckboxes == 1) ? sameLine : true );
```

Comment:
Why do we need to move the cursor? Can't you just render the last item on the same line?